### PR TITLE
Updates units.py for the new api

### LINKED ISSUE #31: WeatherLink Custom Integration - 2023.3.0
I updated my Home Assistant to 2023.3.0...I am now having an issue/error in the logs. Anyone else having this issue? Didn't have this issue on 2023.2.5. Any thoughts?

Logger: homeassistant.config_entries
Source: custom_components/weatherlink/units.py:213
Integration: WeatherLink (documentation, issues)
First occurred: 11:12:07 PM (1 occurrences)
Last logged: 11:12:07 PM

Error setting up entry WeatherLink 001D0A714D87 for weatherlink
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/weatherlink/init.py", line 165, in async_setup_entry
    await setup_coordinator(hass, entry)
  File "/config/custom_components/weatherlink/init.py", line 156, in setup_coordinator
    coordinator = await WeatherLinkCoordinator.build(
  File "/config/custom_components/weatherlink/init.py", line 145, in build
    await coordinator.initalize(session, entry)
  File "/config/custom_components/weatherlink/init.py", line 92, in initalize
    await self.update_config(self.hass, entry)
  File "/config/custom_components/weatherlink/init.py", line 71, in update_config
    self.units = get_unit_config(hass, entry)
  File "/config/custom_components/weatherlink/units.py", line 213, in get_unit_config
    if hass.config.units.name == CONF_UNIT_SYSTEM_IMPERIAL:
AttributeError: 'UnitSystem' object has no attribute 'name'

### PR DESCRIPTION
**Pull request by xdecock**
https://github.com/mrk-its/homeassistant-blitzortung/issues/66 as described here.

Should fix #31

### CODE REVIEW

**Comment by NRutt2005** [comment]
@xdecock does this patch solve the issue that came up in Home Assistant 2023.3.0 as I stated in the hass-weatherlink issues section of this repository?

**Comment by NRutt2005** [comment]
@xdecock I've tried changing line 213 and now I'm getting the error in my log. Any suggestions? This is the error I get after updating again to 2023.3.x. Thank you for your assistance!

This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/weatherlink/units.py:213
Integration: WeatherLink (documentation, issues)
First occurred: 5:49:41 PM (1 occurrences)
Last logged: 5:49:41 PM

Error setting up entry WeatherLink 001D0A714D87 for weatherlink
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/weatherlink/__init__.py", line 165, in async_setup_entry
    await setup_coordinator(hass, entry)
  File "/config/custom_components/weatherlink/__init__.py", line 156, in setup_coordinator
    coordinator = await WeatherLinkCoordinator.build(
  File "/config/custom_components/weatherlink/__init__.py", line 145, in build
    await coordinator.__initalize(session, entry)
  File "/config/custom_components/weatherlink/__init__.py", line 92, in __initalize
    await self.__update_config(self.hass, entry)
  File "/config/custom_components/weatherlink/__init__.py", line 71, in __update_config
    self.units = get_unit_config(hass, entry)
  File "/config/custom_components/weatherlink/units.py", line 213, in get_unit_config
    if hass.config.units is IMPERIAL_SYSTEM:
NameError: name 'IMPERIAL_SYSTEM' is not defined

**Comment by xdecock** [comment]
Yes top of the file addapt the imported constant

**Comment by NRutt2005** [comment]
Sorry, I'm not following when you say "top of the file addapt the imported constant". Could you explain a little bit further for me? Thank you for your help!

**Comment by xdecock** [comment]
> Sorry, I'm not following when you say "top of the file addapt the imported constant". Could you explain a little bit further for me? Thank you for your help!

I've just updated the mr to make it clearer i hope it works i'm on my phone

**Comment by NRutt2005** [comment]
Looks like it's getting closer to working again. The latest error I get now will the two changes. Looks like something in the const.py file needs to be changed? I'm sorry, I'm just good with any sort of coding. I know just enough to get me in trouble as I am in now.

Logger: homeassistant.setup
Source: setup.py:213
First occurred: 12:16:04 AM (1 occurrences)
Last logged: 12:16:04 AM

Setup failed for custom integration weatherlink: Unable to import component: cannot import name 'IMPERIAL_SYSTEM' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const.py)

**Comment by xdecock** [comment]
Ok, this file is fetch from my home assistant, that's not throwing error anymore, and imperial is now deprecated also, so i took the liberty to update this file.

**Review by NRutt2005 — APPROVED** [review]
@siku2 and @ppetru. #32 Can be added as a master v 0.5.4 for fixing issues experience after updating Home Assistant to version 2023.3.x. This patch fixed my issues seen in the conversations with @xdecock. I was able to successfully update to 2023.3.x with no issues after patching my WeatherLink integration with this updated code. Thank you @xdecock for your help fixing this issue!

**Comment by groetg** [comment]
Can you explained what you have changed? I have changed units.py line 213 into hass.config.units.name is IMPERIAL_SYSTEM:
Now I got other errors and is weatherlink addon still not working...

**Comment by xdecock** [comment]
Check the last version of the file in the merge request running smoothly
for days now

**Comment by janssjay** [comment]
Hey team, I have limited experience with Custom Integrations and github.   
Would someone please detail (in layman's terms) the correct steps for finding and applying this patch and upgrading to v0.5.4?
Thanks!

**Comment by NRutt2005** [comment]
@groetg What I did, if I remember correctly, I just copied all 216 lines of code from the Updates unit.py for the new api #32  under "Commits" and clicked in "Update to avoid imperial system being on deprecation list" to see the code. I copied the code and then pasted it in the /config/custom_components/weatherlink/units.py file.

Since doing so, this new code has fixed all my issues. I was hoping that the new code would be pushed out on a release...but I haven't heard from either of the contributors for this repository. I believe one of them has to push it out as a new release. For now, this is the only way I know to make this work.

**Comment by groetg** [comment]
I changed in home assistant / HACS the repository to the version of xdecock (https://github.com/xdecock/hass-weatherlink) and now the weatherlink is active again! Thanx
